pinecone/core/grpc/index_grpc.py

```diff
@@ -264,37 +298,142 @@ def _vector_transform(item):
                 return GRPCVector(id=id, values=values, metadata=dict_to_proto_struct(metadata) or {})
             raise ValueError(f"Invalid vector value passed: cannot interpret type {type(item)}")
 
-        request = UpsertRequest(vectors=list(map(_vector_transform, vectors)), **kwargs)
         timeout = kwargs.pop('timeout', None)
+
+        request = UpsertRequest(vectors=list(map(_vector_transform, vectors)), **args_dict, **kwargs)
         if async_req:
             future = self._wrap_grpc_call(self.stub.Upsert.future, request, timeout=timeout)
             return PineconeGrpcFuture(future)
         else:
             return self._wrap_grpc_call(self.stub.Upsert, request, timeout=timeout)
 
-    def delete(self, *args, async_req=False, **kwargs):
-        _filter = dict_to_proto_struct(kwargs.pop('filter', None))
-        filter_param = {}
-        if _filter:
-            filter_param['filter'] = _filter
-        request = DeleteRequest(*args, **kwargs, **filter_param)
+    def delete(self,
+               ids: Optional[List[str]] = None,
+               delete_all: Optional[bool] = None,
+               namespace: Optional[str] = None,
+               filter: Optional[Dict[str, Union[str, float, int, bool, List, Dict]]] = None,
```

Comment:
```suggestion
               filter: Optional[Dict[str, Union[str, float, int, bool, list, dict]]] = None,
```
You need to use the capital letter `Dict` only for composition (e.g. `Dict[str, str]`).  
For defining "any dict" in `typing` terms, the lower case `dict` is more correct

Reply:
Didn't know that, cool